### DISCUSSION TITLE
Robot Not Loading - The provided path is null, does not exist or cannot be accessed.

### DISCUSSION THREAD

**Discussion by Sophiaon20** [visose/Robots#155] [Q&A] [answered]
Hi my robot load component all of a sudden stopped working for Penn 4600 + Track. I've been using this robot component since August without any issue but now its giving me an error of 1. The provided path is null, does not exist or cannot be accessed.
I triple checked all my robot libraries, the xml file for Penn track, I even deleted and re uploaded a fresh library but its still not working.
Anyone experienced the same issue?

**Comment by visose** [comment]
Hi @Sophiaon20, could you attach the .xml and .3dm files for the robot library you are using?
Could you also mention what version of Robots and Rhino you're using (and if Win or Mac).

You can find the version of **Robots** by opening **Grasshopper**, then Help > About > click on the top left small black square > double click on the **Robots** icon > there should be a number at the top right.

Or you can place some **Robots** components in the canvas, and just run the `GrasshopperPluginList` command in **Rhino**.

**Reply by Sophiaon20** [reply]
I'm using  Windows - rhino 7. Robots Version 1.1.10.
[https://drive.google.com/drive/folders/1_g6w7aJOcO-bJVznQtIaRgx-P7N5Q8mj?usp=sharing](url) the xml and dm file
Thanks

**Reply by visose** [reply]
Just by looking at the file names, it could be because a recent version had a change that required both the .xml and .3dm files to have the exact same name (except for the extension). I'll double check to see if its this.

This change was done because when people had multiple libraries, if different robot cells from different libraries had the same model of mechanism, it could load the wrong one. Most times this was unnoticed because the geometry of the robot was the same, but sometimes they were customized.

**Reply by visose** [reply]
Just to confirm that this was the issue (rename the files to Penn.3dm and Penn.xml, and it will work).
I'll add a better error message for this...

**Reply by Sophiaon20** [reply]
I renamed the files, still doesn't work.
As of one hour ago, not only does the Penn IRB4600 +track, but the Penn-IRB4600+60 doesn't work either.

**Reply by visose** [reply]
Could you post a screenshot of your `Documents\Robots\` folder?

**Reply by Sophiaon20** [reply]
![documentsrobots](https://user-images.githubusercontent.com/66277309/163691897-9c2766a6-9d04-4867-8b45-bdc402172fcb.JPG)

**Reply by visose** [reply]
As you can see with other files there, you have the file extensions hidden, so it's likely your `Penn.xml` file is actually named `Penn.xml.xml`. 
Same thing with `Penn.3dm`. Rename both to just `Penn` and see what happens.

I recommend also deleting everything there except the files you need (the two Penn files)

**Reply by Sophiaon20** [reply]
Thats how I initially had them. I renamed them back - still the same error.

**Reply by Sophiaon20** [reply]
Only one of the four Penn robots actually loads

**Reply by visose** [reply]
The `Penn Track.xml` file that you linked above contains only two robot cells. It works correctly when the files are named the same. I guess the other two robots you mention come from the `Penn.xml` file that I see in the screenshot but you didn't attach.
If you can share it too, I'll merge them and upload it as a single library to the repo. 

Afterwards you just need to delete everything in that folder and use the `Libraries` button in the `Load RobotSystem` component to add the Penn robots.

**Reply by Sophiaon20** [reply]
uploaded

**Comment by visose** [comment, marked as answer]
Would you like me to add the library to the GitHub repo? So it shows up here:

<img width="608" alt="image" src="https://user-images.githubusercontent.com/5129614/163691340-34a689be-6dfc-4f10-810b-35dd10a0df15.png">

This will make it easy to install by other users of the same robots without having to manually copy files.

**Reply by Sophiaon20** [reply]
Sure, i will try that.

**Reply by visose** [reply]
The Penn library should show up now in this panel. Remember to first delete all the files inside your `Documents\Robots` folder

It contains 3 robots (you mentioned 4 but the IRB120 was duplicated appearing in both xml files).

<img width="464" alt="image" src="https://user-images.githubusercontent.com/5129614/163693681-c2080ef5-bdd3-4791-a052-a4c4724588de.png">

**Reply by Sophiaon20** [reply]
Thankyou! Its working now!
